src/Common/Throttler.h

```diff
@@ -10,25 +10,26 @@
 namespace DB
 {
 
-/** Allows you to limit the speed of something (in entities per second) using sleep.
-  * Specifics of work:
-  *  Tracks exponentially (pow of 1/2) smoothed speed with hardcoded window.
-  *  See more comments in .cpp file.
-  *
-  * Also allows you to set a limit on the maximum number of entities. If exceeded, an exception will be thrown.
+/** Allows you to limit the speed of something (in tokens per second) using sleep.
+  * Implemented using Token Bucket Throttling algorithm.
+  * Also allows you to set a limit on the maximum number of tokens. If exceeded, an exception will be thrown.
   */
 class Throttler
 {
 public:
-    explicit Throttler(size_t max_speed_, const std::shared_ptr<Throttler> & parent_ = nullptr)
-            : max_speed(max_speed_), limit_exceeded_exception_message(""), parent(parent_) {}
+    Throttler(size_t max_speed_, size_t max_burst_, const std::shared_ptr<Throttler> & parent_ = nullptr)
+        : max_speed(max_speed_), max_burst(max_burst_), limit_exceeded_exception_message(""), tokens(max_burst), parent(parent_) {}
```

Comment:
Shouldn't we limit `max_burst` somehow? by `max_speed` for example...

Reply:
No, it is independent, zero and very large values also can make sense in some scenarios. I'm going to add them in config in separate PR